packages/webapp/src/containers/Animals/SingleAnimalView/utils.ts

```diff
@@ -30,37 +30,35 @@ const excludeKeys = [
 
 type AnimalBatchKey = keyof (Animal | AnimalBatch);
 
-export const findMissingKeys = (
+export const addNullstoMissingFields = (
   updatedAnimalOrBatch: Partial<Animal | AnimalBatch>,
   originalAnimalOrBatch: Partial<Animal | AnimalBatch>,
-): AnimalBatchKey[] => {
-  const updatedAnimalOrBatchKeys = Object.keys(updatedAnimalOrBatch);
-  const originalAnimalOrBatchKeys = Object.keys(originalAnimalOrBatch);
+): Partial<Animal | AnimalBatch> => {
+  const formattedObject = { ...updatedAnimalOrBatch };
 
-  const missingKeys = originalAnimalOrBatchKeys.filter((key) => {
-    const value = originalAnimalOrBatch[key as AnimalBatchKey];
-    return (
-      !updatedAnimalOrBatchKeys.includes(key) &&
-      value &&
-      (!Array.isArray(value) || value.length > 0) &&
-      !excludeKeys.includes(key)
-    );
-  });
+  const updatedKeys = Object.keys(updatedAnimalOrBatch) as AnimalBatchKey[];
+  const originalKeys = Object.keys(originalAnimalOrBatch) as AnimalBatchKey[];
 
-  return missingKeys as AnimalBatchKey[];
-};
+  const isMissing = (key: AnimalBatchKey): boolean => {
+    const originalValue = originalAnimalOrBatch[key];
 
-export const addNullstoMissingFields = (
-  animalOrBatch: Partial<Animal | AnimalBatch>,
-  missingFields: AnimalBatchKey[],
-): Partial<Animal | AnimalBatch> => {
-  const updatedObject = { ...animalOrBatch };
+    const isKeyMissing = !updatedKeys.includes(key) || updatedAnimalOrBatch[key] === undefined;
```

Comment:
I think `updatedKeys.includes(key)` should be `(key in updatedAnimalOrBatch)` for better performance :)

The refactoring itself looks good!